core/src/test/java/org/kiwix/kiwixmobile/core/search/viewmodel/ZimSearchResultGeneratorTest.kt

```diff
@@ -86,4 +86,44 @@ internal class ZimSearchResultGeneratorTest {
       .isEqualTo(listOf(ZimSearchResultListItem(validTitle)))
     verify { zimReaderContainer.searchSuggestions(searchTerm, 200) }
   }
+
+  @Test
+  internal fun `search term is moved to top of search results`() {
+    val searchTerm = "a"
+    every { sharedPreferenceUtil.prefFullTextSearch } returns false
+    every { zimReaderContainer.getNextSuggestion() } returnsMany listOf(
+      mockk<SearchSuggestion>().also { every { it.title } returns "android" },
+      mockk<SearchSuggestion>().also { every { it.title } returns "a" },
+      mockk<SearchSuggestion>().also { every { it.title } returns "abc" },
+      null
+    )
+    assertThat(zimSearchResultGenerator.generateSearchResults(searchTerm))
+      .isEqualTo(
+        listOf(
+          ZimSearchResultListItem("a"),
+          ZimSearchResultListItem("android"),
+          ZimSearchResultListItem("abc")
+        )
+      )
+    verify { zimReaderContainer.searchSuggestions(searchTerm, 200) }
+  }
+
+  @Test
+  internal fun `invalid search term does not appear in results`() {
```

Comment:
This test I would judge unnecessary because there is no longer code that "inserts a search result"

Reply:
I added that test, not only despite, but because of the exact reason you mentioned.

This test in its current form did not appear in the beginning. It appeared when I removed the search-terms-always-in-results thing. It was a form of test driven development to ensure I did indeed remove it. Currently it's just a way of tracking whether the search term should or should not appear in the search results after repeatedly deciding and changing our minds.

In the beginning, the corresponding test was the polar opposite of this one. Its title was the negation---"suggestion results are sorted with invalid search term added at top".

Reply:
but it is at odds with the functionality that exists, the first test proves a) a search result moves to the top b) items not matching it are unaffected

which is both paths of `sortByDescending{ it == searchTerm }`

While this code may have been useful on the journey I don't think it should exist now. Testing what your code _doesn't_ do is a slippery hill and leads to silliness like
```
fun `when searchTerm is honkhonk do not return list of famous clowns`
```
Obviously exaggerated for effect.

Tests also act as documentation and if I see something like that it sets off warning bells and I might go scouring the git history for a clown enthusiast.

Reply:
@danielzgtg I am happy to get this merged if we remove this test/you convince me this test is necessary.

This PR has been marked stale and we typically close a PR after 2 successive stale markings.

Reply:
I was busy at my job. Also see my comment closing this

Reply:
I was going to remove the test, but before I could do that, I decided that this PR should be closed.